### PR TITLE
Fix/itf

### PR DESCRIPTION
**Pull request by Solonets**
### Description of the Change

FIX: https://soramitsu.atlassian.net/browse/IR-974

### Benefits

now checkProposal and checkBlock in Integration Test Framework work

### Possible Drawbacks 

None

### CODE REVIEW

File: test/integration/pipeline/tx_pipeline_integration_test.cpp [hunk 1]
```diff
@@ -187,6 +187,7 @@ TEST(PipelineIntegrationTest, SendQuery) {
  * @given some user
  * @when sending sample AddAssetQuantity transaction to the ledger
  * @then receive STATELESS_VALIDATION_SUCCESS status on that tx
+ * @then wait for proposal and block
```
Comment (l4l) [inline test/integration/pipeline/tx_pipeline_integration_test.cpp]: Multiple then here, use just `and` instead

File: test/integration/pipeline/tx_pipeline_integration_test.cpp [hunk 2]
```diff
@@ -199,13 +200,21 @@ TEST(PipelineIntegrationTest, SendTx) {
...
+  auto checkStatelessValid = [](auto &status) {
     ASSERT_NO_THROW(
         boost::get<shared_model::detail::PolymorphicWrapper<
             shared_model::interface::StatelessValidTxResponse>>(status.get()));
   };
+  auto checkProposal = [](auto &proposal) {
+    ASSERT_EQ(proposal->transactions.size(), 1);
+  };
+  auto checkBlock = [](auto &block) {
+    ASSERT_EQ(block->transactions.size(), 0);
```
Comment (l4l) [inline test/integration/pipeline/tx_pipeline_integration_test.cpp]: What's the problem with the commit?

File: test/framework/integration_framework/integration_test_framework.hpp [hunk 3]
```diff
@@ -176,9 +180,8 @@ namespace integration_framework {
       ObjectType &ref_for_insertion,
       const WaitTime &wait,
       const std::string &error_reason) {
-    if (!queue.try_pop(ref_for_insertion)) {
-      std::this_thread::sleep_for(wait);
-    }
+    std::unique_lock<std::mutex> lk(queue_mu);
+    queue_cond.wait_for(lk, wait, [&]() { return not queue.empty(); });
```
Comment (l4l) [inline test/framework/integration_framework/integration_test_framework.hpp]: What's the point of the unused predicate? Maybe assert is missing
also `()` are redundant

File: test/framework/integration_framework/integration_test_framework.cpp [hunk 4]
```diff
@@ -96,7 +98,11 @@ namespace integration_framework {
           commit_observable.subscribe([this](auto committed_block) {
             block_queue_.push(
                 std::make_shared<iroha::model::Block>(committed_block));
+            log_->info("block");
+            queue_cond.notify_all();
```
Comment (l4l) [inline test/framework/integration_framework/integration_test_framework.cpp]: Why is that the same `cv` that is used for a proposal? I think they should be different

Comment (l4l) [reply test/framework/integration_framework/integration_test_framework.cpp]: E.g in theory there can be multiple proposals without the commits but `checkProposal(..).checkBlock(..)` will succeed

Comment (Solonets) [reply test/framework/integration_framework/integration_test_framework.cpp]: @l4l as `checkProposal` and `checkBlock` are called sequentially in one thread there are no reason to use different `cv` as they going to block each other any way

File: test/framework/integration_framework/integration_test_framework.hpp (hunk 3 above)
Comment (Solonets) [reply test/framework/integration_framework/integration_test_framework.hpp]: @l4l wat?

File: test/integration/pipeline/tx_pipeline_integration_test.cpp (hunk 2 above)
Comment (Solonets) [reply test/integration/pipeline/tx_pipeline_integration_test.cpp]: @l4l no such user

Comment (Solonets) [reply test/integration/pipeline/tx_pipeline_integration_test.cpp]: @l4l ask the one who wrote this test

Comment (Solonets) [reply test/integration/pipeline/tx_pipeline_integration_test.cpp]: UPD: no permisions

File: test/integration/pipeline/tx_pipeline_integration_test.cpp (hunk 1 above)
Comment (kamilsa) [reply test/integration/pipeline/tx_pipeline_integration_test.cpp]: Please use and as @l4l suggested

**Pull request merged**